### PR TITLE
Make typescript an optional dependency

### PR DESCRIPTION
**Pull request by ndelangen**
if it fails, the plugin does nothing

Hello @joshwooding I decided to sink my teeth into this just to see if I could bite the optional dependency for typescript off.

This is what I came up with.
I understand you might not be a fan, It might solve an issue we're facing at Storybook.

See this PR more as an conversation opener.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,10 +51,14 @@
...
     "vitest": "^0.2.5"
   },
-  "keywords": [
-    "vite",
-    "vite-plugin",
-    "react-docgen-typescript"
-  ],
+  "peerDependencies": {
+    "typescript": ">= 4.3.x",
+    "vite": "^3.0.0"
```
Comment (IanVS) [inline package.json:56]: We should open up the vite dependency to include ^4 as well.  Can do it in a separate PR if we want to keep this PR scoped, but it should be done.

File: src/index.ts [hunk 2]
```diff
@@ -140,27 +29,61 @@ export default function reactDocgenTypescript(config: Options = {}): Plugin {
...
+        }
+
+        return generateDocgenCodeBlock({
+          filename: id,
+          source: src,
+          componentDocs,
+          ...generateOptions,
+        });
+      } catch (e) {
+        return src;
```
Comment (IanVS) [inline src/index.ts:85]: Should we print a warning if this fails?  Might be good to alert the user that it failed.

**Review by IanVS — APPROVED** [review]
Seems like a reasonable approach.

Maybe we should add a note in the README that TypeScript should be installed.

**Comment by joshwooding** [comment]
Happy with this PR. Seems like it's the only way to fix the pnp issues

File: src/index.ts (hunk 2 above)
Comment (ndelangen) [reply src/index.ts:85]: I'm not sure, if so, it should only log once total, not once for every file.

Does vite have some api for reporting problems? If so that would be a lot nicer than rolling our own "dump a bunch of crap" into the terminal.

File: package.json (hunk 1 above)
Comment (joshwooding) [reply package.json:56]: We should do it in another PR

**Comment by ndelangen** [comment]
Let's do an alpha release and test the change in the wild before committing to the solution?

**Comment by joshwooding** [comment]
@ndelangen [0.0.0-20221218231544](https://www.npmjs.com/package/@joshwooding/vite-plugin-react-docgen-typescript/v/0.0.0-20221218231544) is available

**Review by joshwooding — APPROVED** [review]
